copy/copy.go

```diff
@@ -813,6 +814,17 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 		return types.BlobInfo{}, errors.Wrap(err, "Error writing blob")
 	}
 
+	switch compressionOperation {
+	case types.Compress:
+		uploadedInfo.MediaType = manifest.DockerV2Schema2LayerMediaType
+	case types.Decompress:
+		uploadedInfo.MediaType = manifest.DockerV2SchemeLayerMediaTypeUncompressed
+	case types.PreserveOriginal:
+		// preserve whatever the destination sets by default
+	default:
+		panic("unexpected compression operation")
```

Comment:
`return fmt.Errorf("Internal error: invalid compression operation %v")` or something like that, please; if it is not possible to cleanly enough redesign the code to make such a path impossible to reach.

Reply:
Will change, thanks. It's more a precaution in case new types are added.